app/routes/_auth.login.tsx

```diff
@@ -1,16 +1,47 @@
 import type { LoaderArgs, ActionArgs } from '@remix-run/node';
 import { redirect } from '@remix-run/node';
 import { Form, Link, useActionData, useNavigation } from '@remix-run/react';
+import { SignInWithGoogleButton } from '~/components/ui/buttons';
 import { Card } from '~/components/ui/containers';
 import { Input } from '~/components/ui/forms';
 import { matchesHash } from '~/modules/database/crypto.server';
 import { db } from '~/modules/database/db.server';
+import { badRequest } from '~/modules/response/response.server';
+import { verifyGoogleToken } from '~/modules/session/google-auth.server';
 import { createUserSession, getUserSession } from '~/modules/session/session.server';
 
 export async function action({ request }: ActionArgs) {
   const form = await request.formData();
   const email = form.get('email');
   const password = form.get('password');
+  const credential = form.get('credential');
+
+  if (typeof credential === 'string') {
+    try {
```

Comment:
Using a `try catch` here because `verifyGoogleToken` could throw an error if the `credential` token is invalid.